test/distributed/fsdp/test_fsdp_use_orig_params.py

```diff
@@ -1048,10 +1049,122 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         fsdp_model(inp)
 
 
+class TestFSDPUseOrigParamsNoSync(FSDPTest):
+    @property
+    def world_size(self) -> int:
+        return 2
+
+    @skip_if_lt_x_gpu(2)
+    def test_no_sync(self):
+        """
+        Tests a basic ``no_sync()`` setup by comparing ``use_orig_params=True``
+        against ``use_orig_params=False``.
+        """
+        self.run_subtests(
+            {
+                "sharding_strategy": [
+                    ShardingStrategy.FULL_SHARD,
+                    ShardingStrategy.SHARD_GRAD_OP,
+                    ShardingStrategy.NO_SHARD,
+                ],
+            },
+            self._test_no_sync,
+        )
+
+    def _test_no_sync(self, sharding_strategy: ShardingStrategy):
+        model = nn.Linear(3, 3, bias=False, device="cuda")
+        fsdp_kwargs = {
+            "sharding_strategy": sharding_strategy,
+        }
+        baseline_model = FSDP(
+            copy.deepcopy(model), use_orig_params=False, **fsdp_kwargs
+        )
+        test_model = FSDP(model, use_orig_params=True, **fsdp_kwargs)
+        baseline_optim = torch.optim.AdamW(baseline_model.parameters(), foreach=True)
+        test_optim = torch.optim.AdamW(test_model.parameters(), foreach=True)
```

Comment:
nit: might be better to just name these model_no_use_orig and model_use_orig so it's clearer which one is which and what we are comparing.

Reply:
Changed.